core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -2507,4 +2507,13 @@ package object config {
       .version("3.5.0")
       .intConf
       .createWithDefault(Int.MaxValue)
+
+  private[spark] val SHUFFLE_SERVER_RECOVERY_DISABLED =
+    ConfigBuilder("spark.shuffle.server.recovery.disabled")
+      .internal()
+      .doc("Set to true for applications that prefer to disable recovery when the External " +
+        "Shuffle Service restarts.")
+      .version("3.5.0")
+      .booleanConf
+      .createWithDefault(false)
```

Comment:
This is specific to yarn and not supported by other resource managers - call it out in the description ?
Same for the documentation update below - make it explicit that it is for yarn mode, and make sure it is in the right section in the doc.

Reply:
if this is only for yarn should we rename to be spark.yarn.shuffle and only put in yarn docs?

Reply:
- I moved the documentation for this config to `running-on-yarn.md`.
- In the security doc, I moved the definition under `Yarn` section. 
PTAL